crates/sui-gateway/src/event_api.rs

```diff
@@ -0,0 +1,96 @@
+use std::collections::BTreeMap;
+use std::sync::Arc;
+
+use jsonrpsee_core::error::SubscriptionClosed;
+use jsonrpsee_core::server::rpc_module::PendingSubscription;
+use jsonrpsee_proc_macros::rpc;
+use serde::Deserialize;
+use serde::Serialize;
+use strum::IntoEnumIterator;
+use strum_macros::EnumIter;
+use tokio::sync::broadcast;
+use tokio::sync::broadcast::Sender;
+use tokio_stream::wrappers::BroadcastStream;
+use tokio_stream::StreamExt;
+use tracing::{debug, warn};
+
+use sui_core::gateway_types::SuiEvent;
```

Comment:
We have multiple struct Event defined! They should be consolidated into https://github.com/MystenLabs/sui/blob/1de042041b110bf75e31b41b886139eefd95cae4/crates/sui-types/src/event.rs#L48 once @velvia 's  #2215 settles dowm

Reply:
We will still need a new type for JSON/readability
We will need 2 `Event` types here, one for internal use - optimised for storage/node to node message and another one for external -- optimised for JSON/readability

Reply:
@patrickkuo is right, but Patrick have a look at the `StoredEvent` struct which I defined in `event_store.rs` in Sui-storage - it's what I intend as the output of EventStore and should be helpful for formulating `SuiEvent`.  In fact I intended to change `SuiEvent` in gateway to model after `StoredEvent` but using JSON types.  WDYT?

Reply:
yup that sounds good